crypto/rand/rand_lib.c

```diff
@@ -25,6 +25,45 @@ static CRYPTO_RWLOCK *rand_meth_lock;
 static const RAND_METHOD *default_RAND_meth;
 static CRYPTO_ONCE rand_init = CRYPTO_ONCE_STATIC_INIT;
 
+#ifdef OPENSSL_RAND_SEED_RDTSC
+/*
+ * Since we get some randomness from the low-order bits of the
+ * high-speec clock, it can help.  But don't return a status since
+ * it's not sufficient to indicate whether or not the seeding was
+ * done.
+ */
+void rand_rdtsc(void)
+{
+    unsigned char c;
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        c = (unsigned char)(OPENSSL_rdtsc() & 0xFF);
```

Comment:
My measurements, using the xor reduction I mentioned, resulted in just over 7.8 bits per byte when processed by the NIST SP800-90B (second draft) estimates.  The 7.8 figure is the best result I've seen from these tests and is on par with /dev/random or RDRAND.  The data pass the small crush suite in TestU01.  They pass the crush test as well but the number of samples is too small to draw a conclusion.  The dieharder tests also require more data than were collected.

For reference, I sampled TSC at about 8 Hz but the results held for at least another order of magnitude increase in the sampling rate.  Given the amount of other work `RAND_poll` is doing, I don't foresee a significant risk of sampling fast enough to cause an issue.

I don't see a problem with crediting up to 7.8 / 8 = **0.975** bytes of randomness using the approach of adding the entire result, this should be an underestimate.  Play it conservatively and reduce this figure a bit.  The 0.5 it is set at currently is very conservative.  A value of **0.8** or **0.9** should still have an adequate margin of safety.  Note this isn't per byte, it is total.

I've tested a variety of x86/64 processors (old and new) running Solaris, Linux and Windows.  The equivalent to RDTSC on Itanium and SPARC produce similar results.  I've not tested other architectures that support similar functionality but probably should.